src/sdk/pynni/nni/compression/torch/speedup/compress_modules.py

```diff
@@ -115,15 +115,18 @@ def replace_conv2d(conv, mask):
     else:
         out_channels_index = mask.output_mask.mask_index[1]
         out_channels = out_channels_index.size()[0]
-
-    _logger.debug("replace conv2d with in_channels: %d, out_channels: %d", in_channels, out_channels)
+    groups = conv.groups
+    if conv.in_channels == conv.out_channels == conv.groups:
+        assert in_channels == out_channels
+        groups = in_channels
```

Comment:
Why do we need such a specific judgment?

Reply:
for depthwise conv layers,  such as  in_channel == out_channel == groups == 32,   if we prune any of the three, for example prune output to 16 channels,  we need to prune input channels and groups to 16 too, otherwise it does not work.

Reply:
this is specifically for depthwise conv, suggest to add comment to mention it.

Reply:
maybe we can make it general?

Reply:
seems it is indeed specific for depthwise conv in which case the number of groups is reduced, in other cases the number is not changed

Reply:
Maybe we can generalize this into two categories:
(1) keep the number of the filters of each group equal, (don't change the number of the group)
(2) just remove the whole group filters(not only for the depth-wise conv)
I'm not sure if I express this clearly, we can set up a quick meeting and discuss this